### PR TITLE
[polyhook2] Update polyhook2

### PR DESCRIPTION
**Pull request by stevemk14ebr**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/polyhook2/vcpkg.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "polyhook2",
-  "version-date": "2023-05-16",
+  "version-date": "2023-08-12",
```
Comment (MonicaLiu0311) [inline ports/polyhook2/vcpkg.json:3]:
```suggestion
  "version-date": "2023-08-11",
```
`REF 71553ebf0d197e73c49365cc3ab2a6aa326b3c37`
This date should preferably be the upstream commit date.

**Comment by MonicaLiu0311** [comment]
All features are tested successfully in the following triplet:
```
x86-windows
x64-windows
x64-windows-static
```

The usage test passed on `x64-windows` (header files found):
```
polyhook2 provides CMake targets:

    # this is heuristically generated, and may not be correct
    find_package(PolyHook_2 CONFIG REQUIRED)
    target_link_libraries(main PRIVATE PolyHook_2::PolyHook_2)
```

File: ports/polyhook2/vcpkg.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "polyhook2",
-  "version-date": "2023-05-16",
+  "version-date": "2023-08-11",
   "port-version": 1,
```
Comment (MonicaLiu0311) [inline ports/polyhook2/vcpkg.json:4]:
```suggestion
```

File: versions/p-/polyhook2.json [hunk 3]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "bc5dee5365ffa6db581bd9680bcc4afcefbb211a",
+      "version-date": "2023-08-11",
+      "port-version": 1
+    },
```
Comment (MonicaLiu0311) [inline versions/p-/polyhook2.json:7]:
```
2023-08-11#1 is declared with bc5dee5365ffa6db581bd9680bcc4afcefbb211a, but the local port has a different SHA 8b5ef7330c17cce8fa0b627027f522e6979027ec.
Please update the port's version fields and then run:
vcpkg x-add-version polyhook2 --overwrite-version
```
At present, the version information of your commit is inconsistent with your local one:

**Comment by MonicaLiu0311** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review".

**Review by MonicaLiu0311 — DISMISSED** [review]

File: versions/p-/polyhook2.json [hunk 4]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "8b5ef7330c17cce8fa0b627027f522e6979027ec",
+      "version-date": "2023-08-11",
+      "port-version": 1
```
Comment (dan-shaw) [inline versions/p-/polyhook2.json:6]: I think this `port-version` should be `0`.

Comment (MonicaLiu0311) [reply versions/p-/polyhook2.json:6]: @stevemk14ebr

**Comment by stevemk14ebr** [comment]
I fixed the version and ran the overwrite command, IDK what's wrong with the CI. This is annoying.

**Comment by stevemk14ebr** [comment]
The merge conflict decided to overwrite the old port version instead of adding a new line. Weird. Can this version stuff be done via a github task instead?

**Review by MonicaLiu0311 — APPROVED** [review]

**Pull request merged**